crates/db_schema/src/source/comment.rs

```diff
@@ -49,16 +54,34 @@ pub struct Comment {
   /// Whether the comment has been distinguished(speaking officially) by a mod.
   pub distinguished: bool,
   pub language_id: LanguageId,
+  #[diesel(select_expression = score(comment::non_1_upvotes, comment::non_0_downvotes))]
+  #[diesel(select_expression_type = score<comment::non_1_upvotes, comment::non_0_downvotes>)]
   pub score: i32,
+  #[diesel(select_expression = coalesce(comment::non_1_upvotes, 1))]
+  #[diesel(select_expression_type = coalesce<sql_types::Integer, comment::non_1_upvotes, i32>)]
   pub upvotes: i32,
+  #[diesel(select_expression = coalesce(comment::non_0_downvotes, 0))]
+  #[diesel(select_expression_type = coalesce<sql_types::Integer, comment::non_0_downvotes, i32>)]
   pub downvotes: i32,
   /// The total number of children in this comment branch.
+  #[diesel(select_expression = coalesce(comment::non_0_child_count, 0))]
+  #[diesel(select_expression_type = coalesce<sql_types::Integer, comment::non_0_child_count, i32>)]
   pub child_count: i32,
   #[serde(skip)]
-  pub hot_rank: f64,
+  #[diesel(select_expression = hot_rank(comment::non_1_upvotes, comment::non_0_downvotes, comment::age))]
+  #[diesel(select_expression_type = hot_rank<comment::non_1_upvotes, comment::non_0_downvotes, comment::age>)]
+  pub hot_rank: f32,
```

Comment:
This means postgres has to run a function for every single row selected, whereas before we had this data cached. Saving space shouldn't be prioritized over performance, especially for ranks.